audformat/core/utils.py

```diff
@@ -1561,3 +1500,63 @@ def union(
     index = index.drop_duplicates()
 
     return index
+
+
+def _convert_single_level_multi_index(
+        objs: typing.List[typing.Union[pd.Index, pd.Series, pd.DataFrame]],
+):
+    r"""Convert single-level pd.MultiIndex to pd.Index.
+
+    If input is a mixture of single-level
+    pd.MultiIndex and pd.Index objects,
+    all objects are converted to pd.Index.
+    Assumes that list is not empty.
+
+    Args:
+        objs: list with objects
+
+    Raises:
+        ValueError: if level and dtypes of objects do not match
+
+    """
+    if not is_index_alike(objs):
+        raise ValueError(
+            'Levels and dtypes of all objects must match, '
+            'see audformat.utils.is_index_alike().'
+        )
+
+    if isinstance(objs[0], pd.Index):
+        is_single_level = objs[0].nlevels == 1
+        is_mix = len(set(isinstance(obj, pd.MultiIndex)
+                         for obj in objs)) == 2
+    else:
+        is_single_level = objs[0].index.nlevels == 1
+        is_mix = len(set(isinstance(obj.index, pd.MultiIndex)
+                         for obj in objs)) == 2
+
+    if is_single_level and is_mix:
+        if isinstance(objs[0], pd.Index):
+            for idx, obj in enumerate(objs):
+                if isinstance(obj, pd.MultiIndex):
+                    objs[idx] = obj.get_level_values(0)
+        else:
+            for idx, obj in enumerate(objs):
+                if isinstance(obj.index, pd.MultiIndex):
+                    objs[idx].index = obj.index.get_level_values(0)
```

Comment:
Maybe

```suggestion                
    indices = [obj if isinstance(obj, pd.Index) else obj.index for obj in objs]
    is_single_level = indices[0].nlevels == 1 
    is_mix = len(set(isinstance(index, pd.MultiIndex)
                     for index in indices)) == 2 

    if is_single_level and is_mix:
        for idx, obj in enumerate(objs):
            if isinstance(obj, pd.MultiIndex):
                objs[idx] = obj.get_level_values(0)
            elif (
                    not isinstance(obj, pd.Index)
                    and isinstance(obj.index, pd.MultiIndex)
            ):
                objs[idx].index = obj.index.get_level_values(0)
```

This is easier to understand at the beginning, and the second part will also work if you provide a mixture of `Index` and `Series`/`DataFrame` objects.